File: base/src/main/java/com/zhangteng/base/adapter/CommonFragmentAdapter.java
package com.zhangteng.base.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;

/**
 * 公共的fragmentPagerAdapter
 * Created by Swing on 2017/11/24.
 */

public class CommonFragmentAdapter extends FragmentPagerAdapter {
    private String[] titles;
    private ArrayList<Fragment> fragments;

    public CommonFragmentAdapter(FragmentManager fm, ArrayList<Fragment> fragments) {
        super(fm);
        this.fragments = fragments;
        this.titles = new String[fragments.size()];
    }

    public CommonFragmentAdapter(FragmentManager fm, String[] titles, ArrayList<Fragment> fragments) {
        super(fm);
        this.titles = titles;
        this.fragments = fragments;
    }

    public ArrayList<Fragment> getFragments() {
        return fragments;
    }

    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return titles[position];
    }

    @Override
    public int getCount() {
        return fragments.size();
    }
}

File: base/src/main/java/com/zhangteng/base/adapter/CommonFragmentAdapter.kt
package com.zhangteng.base.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import java.util.*

/**
 * 公共的fragmentPagerAdapter
 * Created by Swing on 2017/11/24.
 */
open class CommonFragmentAdapter : FragmentPagerAdapter {
    private var titles: Array<String?>?
    private var fragments: ArrayList<Fragment>

    constructor(fm: FragmentManager, fragments: ArrayList<Fragment>) : super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
        this.fragments = fragments
        titles = arrayOfNulls<String?>(fragments.size)
    }

    constructor(fm: FragmentManager, titles: Array<String?>?, fragments: ArrayList<Fragment>) : super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
        this.titles = titles
        this.fragments = fragments
    }

    fun getFragments(): ArrayList<Fragment> {
        return fragments
    }

    override fun getItem(position: Int): Fragment {
        return fragments[position]
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return titles?.get(position)
    }

    override fun getCount(): Int {
        return fragments.size
    }
}